msal/managed_identity.py

```diff
@@ -288,6 +289,29 @@ def _scope_to_resource(scope):  # This is an experimental reasonable-effort appr
     return scope  # There is no much else we can do here
 
 
+APP_SERVICE = object()
+AZURE_ARC = object()
+CLOUD_SHELL = object()
+MACHINE_LEARNING = object()
+SERVICE_FABRIC = object()
+DEFAULT_TO_VM = object()  # Unknown environment; default to VM; you may want to probe
+def get_managed_identity_source():
```

Comment:
Not very familiar with the python but shouldn't there be unit tests for this flow?

Reply:
Did not aim for 100% [test coverage](https://stackoverflow.com/questions/90002/what-is-a-reasonable-code-coverage-for-unit-tests-and-why). But, yeah, new test cases is added now.